dmutils/email.py

```diff
@@ -231,3 +232,95 @@ def decode_invitation_token(encoded_token):
 
 def token_created_before_password_last_changed(token_timestamp, user_timestamp):
     return token_timestamp < user_timestamp
+
+
+class DMNotifyClient(object):
+    """Digital Marketplace wrapper around the Notify python client."""
+
+    client_class = NotificationsAPIClient
+
+    def __init__(self, api_key, base_url='https://api.notifications.service.gov.uk', logger=None):
+        """Set up logging and mail client."""
+        self.logger = logger or current_app.logger
+        self.client = self.client_class(api_key, base_url)
+
+    @staticmethod
+    def get_reference(email_address, template_id, personalisation, time=None, set_time=True, **kwargs):
+        """
+        Method to return the standard reference given the variables the email is sent with.
+
+        :param email_address: Emails recipient
+        :param template_id: Emails template ID on Notify
+        :param personalisation: Template parameters
+        :param time: Generally not wise to set this. Spoofs time that the email was sent.
+        :param set_time: Flag, whether or not to set the time in the reference.
+        :param kwargs: Extra data passed to the reference eg. {'notes': 'Manual resend'}
+        :return: String json 'reference' to be passed to client.send_email_notification or self.send_email
+        """
+        if time is None and set_time:
```

Comment:
To check for the same email you need to specify the email address, template id and the 'personalisation'. Adding time ensures that the reference is unique. In my view the reference should really represent a 'send' instance rather than the content of the email and the most useful way I could think of to make a unique reference was just dumping in the attributes we used to create the email.